src/react-cancelable.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Observable } from "rx";
+
+const inTestingEnv = typeof jasmine !== "undefined";
+
+export function Cancelable() {
+	return function decorator(DecoratedComponent) {
+		return class Disposable extends React.Component {
+			disposables = [];
+			render() {
+				return (
+					<DecoratedComponent
+						{...this.props}
+						cancelWhenUnmounted={this.cancelWhenUnmounted}
+						cancelAllSubscriptions={this.cancelAllSubscriptions}
+					/>
+				);
+			}
+			componentWillUnmount() {
+				this.cancelAllSubscriptions();
+			}
+			cancelWhenUnmounted = cancelWhenUnmounted.bind(this);
+			cancelAllSubscriptions = cancelAllSubscriptions.bind(this);
+		};
+	};
+}
+
+export function cancelableModal(El, props = {}) {
+	let el;
+
+	return Observable.create(observer => {
+		el = document.createElement("div");
+		document.body.appendChild(el);
+
+		ReactDOM.render(
+			<El {...props} close={observer.onCompleted} save={observer.onNext} />,
+			el
+		);
+	}).finally(() => {
+		ReactDOM.unmountComponentAtNode(el);
```

Comment:
A (maybe opinionated?) behavior in the implementation here is that the modal will be unmounted and removed from the dom whenever it closes OR errors. [rx's finally operator](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/finally.md) gets called in both situations. The other option would be to use [tapOnCompleted](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/dooncompleted.md) which does not unmount the modal whenever an error occurs. Or maybe it could be an option provided to `cancelableModal` that defaults to one but allows for the other?

Couple other things to think about:
1. If the modal throws an error during the first render, the observable onErrors. Is that desirable?
2. If a subsequent render of the modal (via setState or forceUpdate) causes an error, should the observable onError? If so, is that a use case for `componentDidCatch()`?